CMakeLists.txt

```diff
@@ -130,20 +130,9 @@ if(ENABLE_PROGRAMS)
     add_subdirectory(programs)
 endif()
 
-# targets for doxygen only work on Unix
-if(UNIX)
-    ADD_CUSTOM_TARGET(apidoc
-        COMMAND mkdir -p apidoc
-        COMMAND cp include/mbedtls/config.h include/mbedtls/config.h.bak
-        COMMAND scripts/config.pl realfull
-        COMMAND doxygen doxygen/mbedtls.doxyfile
-        COMMAND mv include/mbedtls/config.h.bak include/mbedtls/config.h
-        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
-
-    ADD_CUSTOM_TARGET(apidoc_clean
-        COMMAND rm -rf apidoc
-        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
-endif(UNIX)
+ADD_CUSTOM_TARGET(apidoc
```

Comment:
Why are we changing the definition of targets in a maintenance branch? They should stay the same across patch revisions.

The purpose may be the same, but we're changing the outputs and meaning of the target from one thing to another.

This may be documentation, but the point remains. We shouldn't do this in a maintenance branch which is supposed to be bug-fixes only.

Reply:
But isn't it a bug that `make apidoc` generates the documentation for the “full” configuration rather than for the current configuration? Especially given that there's no obvious way to generate the documentation for the current configuration.

Reply:
You could call it a bug, but I don't think we should make a behavioural change on a maintained branch, just on principle.

I think it was @pjbakker's intention that the documentation be generated this way, and I think we should leave it as is.

Reply:
I accept @mpg's argument that we're just reverting it back to the way it was in 2.1.3